src/common/storage/src/parquet2.rs

```diff
@@ -87,35 +87,36 @@ async fn read_parquet_metas_batch(
 pub async fn read_parquet_metas_in_parallel(
     op: Operator,
     file_infos: Vec<(String, u64)>,
-    thread_nums: usize,
-    permit_nums: usize,
+    num_threads: usize,
     max_memory_usage: u64,
 ) -> Result<Vec<FileMetaData>> {
-    let batch_size = 100;
-    if file_infos.len() <= batch_size {
-        read_parquet_metas_batch(file_infos, op.clone(), max_memory_usage).await
-    } else {
-        let mut chunks = file_infos.chunks(batch_size);
+    let num_files = file_infos.len();
+    let mut tasks = Vec::with_capacity(num_threads);
 
-        let tasks = std::iter::from_fn(move || {
-            chunks.next().map(|location| {
-                read_parquet_metas_batch(location.to_vec(), op.clone(), max_memory_usage)
-            })
-        });
+    // Equally distribute the tasks
+    for i in 0..num_threads {
+        let begin = num_files * i / num_threads;
+        let end = num_files * (i + 1) / num_threads;
+        if begin == end {
```

Comment:
if num_threads = 8,  num_files= 2.
we should start 2 threads 
but this code result in 8 threads but only the first has 2 files.